cmake/projects/OpenSSL/schemes/url_sha1_openssl.cmake.in

```diff
@@ -50,11 +52,17 @@ if(ANDROID)
     hunter_user_error("Could not find android architecture. Please set ANDROID_ARCH_NAME in your toolchain or use CMake 3.7+")
   endif()
   set(configure_command
-      # Ignored. Prevents script from checking host uname
-      RELEASE=2.6.37
-      SYSTEM=android
-      ARCH=${ANDROID_SSL_ARCH}
-      ${configure_command})
+      ${configure_command} ${CMAKE_C_FLAGS})
+
+  # Latest OpenSSL (1.1.1) still does not have correct android targets, so set generic with our CFLAGS
+  string(COMPARE EQUAL "${ANDROID_SSL_ARCH}" "arm64" _is_arm64)
+  string(COMPARE EQUAL "${ANDROID_SSL_ARCH}" "mips64" _is_mips64)
+  string(COMPARE EQUAL "${ANDROID_SSL_ARCH}" "x86_64" _is_x8664)
+  if (_is_arm64 OR _is_mips64 OR _is_x8664)
+    set(configure_opts "linux-generic64")
```

Comment:
It's okay to introduce workaround in Hunter but this workaround should be focused and linked to opened upstream bugs. We must understand what is the right approach and what is the "use hack until bug officially fixed". I'm just trying to avoid "this nonstandard build work for me" topic because as I already said next time some developer will report issue with your approach and will ask quite fair question why I have accepted the removing of official build instructions.